helm-charts/charts/kube-starrocks/charts/starrocks/templates/_helpers.tpl

```diff
@@ -240,3 +240,332 @@ the first 8 digits are taken, which will be used as the annotations for pods.
+*/}}
+{{- define "starrockscluster.cn.serviceAccount" -}}
+{{- if .Values.starrocksCnSpec.serviceAccount -}}
+{{- .Values.starrocksCnSpec.serviceAccount -}}
+{{- else if .Values.starrocksCluster.componentValues.serviceAccount -}}
+{{- .Values.starrocksCluster.componentValues.serviceAccount -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Get the value of the imagePullSecrets field in the starrocksFESpec
+*/}}
+{{- define "starrockscluster.fe.imagePullSecrets" -}}
+{{- if .Values.starrocksFESpec.imagePullSecrets -}}
+{{- toYaml .Values.starrocksFESpec.imagePullSecrets -}}
+{{- else if .Values.starrocksCluster.componentValues.imagePullSecrets -}}
+{{- toYaml .Values.starrocksCluster.componentValues.imagePullSecrets -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Get the value of the imagePullSecrets field in the starrocksBeSpec
+*/}}
+{{- define "starrockscluster.be.imagePullSecrets" -}}
+{{- if .Values.starrocksBeSpec.imagePullSecrets -}}
+{{- toYaml .Values.starrocksBeSpec.imagePullSecrets -}}
+{{- else if .Values.starrocksCluster.componentValues.imagePullSecrets -}}
+{{- toYaml .Values.starrocksCluster.componentValues.imagePullSecrets -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Get the value of the imagePullSecrets field in the starrocksCnSpec
+*/}}
+{{- define "starrockscluster.cn.imagePullSecrets" -}}
+{{- if .Values.starrocksCnSpec.imagePullSecrets -}}
+{{- toYaml .Values.starrocksCnSpec.imagePullSecrets -}}
+{{- else if .Values.starrocksCluster.componentValues.imagePullSecrets -}}
+{{- toYaml .Values.starrocksCluster.componentValues.imagePullSecrets -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Get the value of the tolerations field in the starrocksFESpec
+*/}}
+{{- define "starrockscluster.fe.tolerations" -}}
+{{- if .Values.starrocksFESpec.tolerations -}}
+{{- toYaml .Values.starrocksFESpec.tolerations -}}
+{{- else if .Values.starrocksCluster.componentValues.tolerations -}}
+{{- toYaml .Values.starrocksCluster.componentValues.tolerations -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Get the value of the tolerations field in the starrocksBeSpec
+*/}}
+{{- define "starrockscluster.be.tolerations" -}}
+{{- if .Values.starrocksBeSpec.tolerations -}}
+{{- toYaml .Values.starrocksBeSpec.tolerations -}}
+{{- else if .Values.starrocksCluster.componentValues.tolerations -}}
+{{- toYaml .Values.starrocksCluster.componentValues.tolerations -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Get the value of the tolerations field in the starrocksCnSpec
+*/}}
+{{- define "starrockscluster.cn.tolerations" -}}
+{{- if .Values.starrocksCnSpec.tolerations -}}
+{{- toYaml .Values.starrocksCnSpec.tolerations -}}
+{{- else if .Values.starrocksCluster.componentValues.tolerations -}}
+{{- toYaml .Values.starrocksCluster.componentValues.tolerations -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Get the value of the nodeSelector field in the starrocksFESpec
+*/}}
+{{- define "starrockscluster.fe.nodeSelector" -}}
+{{- if .Values.starrocksFESpec.nodeSelector -}}
```

Comment:
this is a complete replace behavior, not the merge together behavior, need explicit this in the values.yaml comments.

Otherwise, the user can define some of the common criteria in compontValues and add additional specific criteria in each XXSpec and expect the final result is the merge-replace output.

The same applies to hostAlias/Labels/affinity ...

Reply:
committed